qml/pages/PollView.qml

```diff
@@ -0,0 +1,229 @@
+                        va[model.id] = !checked
+                        voteTracker = new Object(va)
+                    } else if (polldata.type === "regular") { // reset uservotes to contain just this
+                        var va = {}
+                        va[model.id] = !checked
+                        voteTracker = new Object(va)
+                    } else { // not supported
+                        console.warn("click in unsupported poll mode")
+                    }
+                }
+            }
+            Row { id: viewText
+                visible: bars.visible
+                width: parent.width - Theme.horizontalPageMargin
+                anchors.horizontalCenter: parent.horizontalCenter
+                opacity: visible ? 1.0 : 0
+                Behavior on opacity { NumberAnimation { } }
+                Label { id: pollText
+                    width: parent.width - pollNum.width
+                    text: html
+                    textFormat: Text.StyledText
+                    wrapMode: Text.Wrap
+                }
+                Label { id: pollNum
+                    text: votes //+ " (" + (votes / polldata.voters * 100).toFixed(1) + "%)"
+                    color: Theme.secondaryColor
+                    anchors.verticalCenter: parent.verticalCenter
+                }
+            }
+            // TODO: chart type *could* be "pie chart"
+            //ProgressCircle { }
+            ProgressBar { id: bars
+                width: parent.width - Theme.horizontalPageMargin
+                anchors.horizontalCenter: parent.horizontalCenter
+                leftMargin: 0
+                rightMargin: 0
+                visible: bars.visible
+                // that truncates, not good for long text.
+                //label: html
+                // large and ugly when used below a text label:
+                // valueText: votes
+                minimumValue: !!polldata.min ? polldata.min - 1 : 0
+                maximumValue: polldata.voters
+                value: votes
+            }
+        }
+        VerticalScrollDecorator {}
+        ViewPlaceholder { id: viewplaceholder }
+        PullDownMenu{
+            MenuItem { id: resetMenu
+                visible: canVote
+                enabled: canSubmit
+                text: qsTr("Reset")
+                onClicked: { voteTracker = new Object({}); populate() }
+            }
+            MenuItem { id: submitMenu
+                visible: canVote
+                enabled: canSubmit
+                text: qsTr("Submit")
+                onClicked: {
+                    // FIXME: there's surely a more javascripty way to do that:
+                    // make an array of ids out of the object with "id" as property name
+                    var opts = []
+                    var ids = Object.keys(voteTracker)
+                    ids.forEach(function(e) { if (data[e]) opts.push(e) })
+                    submitPoll(key, postid, polldata.name, options)
```

Comment:
190: ... if (data...
seems to fail here, just pushing opts worked on a single poll vote, needs more testing on multi-choice

191: submitPoll(..., options) 
typo, should be opts